tasks/bridge_configuration.yml

```diff
@@ -48,35 +54,41 @@
   become: true
   template:
     src: 'rule_{{ ansible_facts.os_family }}.j2'
-    dest: '{{ interfaces_net_path[ansible_facts.os_family|lower] }}/rule-{{ item.device }}'
+    dest: '{{ interfaces_net_path }}/rule-{{ item.device }}'
   with_items: '{{ interfaces_bridge_interfaces }}'
-  when: item.rules is defined and ansible_facts.os_family == 'RedHat'
+  when:
+    - item.rules is defined
+    - ansible_facts.os_family == 'RedHat'
+    - not interfaces_use_nmconnection
   register: bridge_rule_add_result
   notify:
     - Bounce network devices
 
 - name: RedHat | Remove configuration files for rhel rule configuration
   become: true
   file:
-    path: '{{ interfaces_net_path[ansible_facts.os_family|lower] }}/rule-{{ item.device }}'
+    path: '{{ interfaces_net_path }}/rule-{{ item.device }}'
     state: absent
   with_items: '{{ interfaces_bridge_interfaces }}'
-  when: item.rules is not defined and ansible_facts.os_family == 'RedHat'
+  when:
+    - item.rules is not defined or interfaces_use_nmconnection
+    - ansible_facts.os_family == 'RedHat'
   register: bridge_rule_del_result
   notify:
     - Bounce network devices
 
 - name: Create the network configuration file for port on the bridge devices
   become: true
   template:
-    src: 'bridge_port_{{ ansible_facts.os_family }}.j2'
-    dest: '{{ interfaces_net_path[ansible_facts.os_family|lower] }}/ifcfg-{{ item.1 }}'
+    src: "bridge_port_{{ interfaces_use_nmconnection | ternary('nmconnection', ansible_facts.os_family) }}.j2"
+    dest: "{{ interfaces_net_path }}/{{ interfaces_use_nmconnection | ternary(item.1 ~ '.nmconnection', 'ifcfg-' ~ item.1) }}"
```

Comment:
we need `mode: "{{ interfaces_use_nmconnection | ternary('0600', omit) }}"` here as well

Reply:
Good catch, thanks :+1: